### PR TITLE
telemetrygateway: add country code to event attributes

### PR DESCRIPTION
**Pull request by bobheadxi**
This change extends the data exported to Telemetry Gateway with inferred client geolocation, added in https://github.com/sourcegraph/sourcegraph/pull/58386, as discussed [here](https://sourcegraph.slack.com/archives/C01PQEVJ5A9/p1700078374822459?thread_ts=1694717740.084639&cid=C01PQEVJ5A9)

## Test plan

CI

### CODE REVIEW

**Comment by bobheadxi** [comment]
Current dependencies on/for this PR:
* main
  * **PR #58386** <a href="https://app.graphite.dev/github/pr/sourcegraph/sourcegraph/58386?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #58388** <a href="https://app.graphite.dev/github/pr/sourcegraph/sourcegraph/58388?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/sourcegraph/sourcegraph/58388?utm_source=stack-comment).

File: internal/telemetrygateway/v1/telemetrygateway.proto [hunk 1]
```diff
@@ -204,4 +204,15 @@ message EventMarketingTracking {
 message EventInteraction {
   // OpenTelemetry trace ID representing the interaction associated with the event.
   optional string trace_id = 1;
+
+  // Reserve entry for client-provided interaction ID: https://github.com/sourcegraph/sourcegraph/pull/58016
+  reserved 2;
```
Comment (bobheadxi) [inline internal/telemetrygateway/v1/telemetrygateway.proto:209]: For https://github.com/sourcegraph/sourcegraph/pull/58016

**Review by unknwon — APPROVED** [review]

**Comment by bobheadxi** [comment]
Merging after https://github.com/sourcegraph/sourcegraph/pull/58515

**Comment by bobheadxi** [comment]
https://github.com/sourcegraph/sourcegraph/pull/58537

**Pull request merged**